### PR TITLE
implement backend integration

### PR DESCRIPTION
**Pull request by tomered**

### CODE REVIEW

File: src/Components/AppMenu/AppMenu.tsx [hunk 1]
```diff
@@ -40,7 +44,15 @@ export default function AppMenu() {
         <AppMenuItem tileType={Tiles.ENDING_TILE} />
         <AppMenuItem tileType={Tiles.BLOCK_TILE} />
         <AppMenuItem tileType={Tiles.EMPTY_TILE} />
-        <MenuItem onClick={() => console.log(graph)}>
+        <MenuItem
+          onClick={() => {
```
Comment (saarbm25A) [inline src/Components/AppMenu/AppMenu.tsx]: don't use anonyms function

File: src/Components/AppTile/AppTile.tsx [hunk 2]
```diff
@@ -21,14 +23,31 @@ const AppTile = ({
...
 
   const handleClick = () => {
     if (!disabled && tileType && position) {
       dispatch(setGraphTile({ i: position.i, j: position.j, tileType }));
     }
   };
-  const bgcolor =
-    graph && position ? COLORS[graph[position.i][position.j]] : color;
+
+  const getBGColor = () => {
```
Comment (saarbm25A) [inline src/Components/AppTile/AppTile.tsx:34]: Add a comment explaining the logic here

**Review by saarbm25A — CHANGES_REQUESTED** [review]

File: src/Components/AppTile/AppTile.tsx [hunk 3]
```diff
@@ -21,14 +23,35 @@ const AppTile = ({
...
+        }
+
+        return false;
+      });
+      // Return the path color
+      if (isPositionInPath) {
+        return COLORS.PATH_TILE;
+      }
+    }
+    // Check if graph and position exists, if so return the position tile color, otherwise return empty tile color
```
Comment (saarbm25A) [inline src/Components/AppTile/AppTile.tsx]: This is simply wrong.

**Review by saarbm25A — CHANGES_REQUESTED** [review]

File: src/Components/AppTile/AppTile.tsx (hunk 3 above)
Comment (tomered) [reply src/Components/AppTile/AppTile.tsx]: what is wrong?

Comment (saarbm25A) [reply src/Components/AppTile/AppTile.tsx]: The comment you wrote incorrectly sums up the logic of that line, specifically the "otherwise" part.

**Review by saarbm25A — APPROVED** [review]

**Pull request merged**